spinalcordtoolbox/mtsat/mtsat.py

```diff
@@ -0,0 +1,151 @@
+# coding: utf-8
+# This is the interface API to compute MTsat
+# Code is based on QMRLab: https://github.com/neuropoly/qMRLab
+# Author: Julien Cohen-Adad
+# Copyright (c) 2018 Polytechnique Montreal <www.neuro.polymtl.ca>
+# About the license: see the file LICENSE.TXT
+
+import os
+import sct_utils as sct
+from msct_image import Image
+import numpy as np
+import math
+
+def compute_mtsat(nii_mt, nii_pd, nii_t1,
+                  tr_mt, tr_pd, tr_t1,
+                  fa_mt, fa_pd, fa_t1,
+                  nii_b1map=None, verbose=1):
+    """
+    Compute MTsat and T1 map based on FLASH scans
+    Parameters
+    ----------
+    nii_mt: msct_image object
+    nii_pd: msct_image object
+    nii_t1: msct_image object
+    tr_mt: TR in ms
+    tr_pd: TR in ms
+    tr_t1: TR in ms
+    fa_mt: flip angle in deg
+    fa_pd: flip angle in deg
+    fa_t1: flip angle in deg
+    nii_b1map: msct_image object of B1+ field map, used to correct the flip angle
+    verbose: 0, 1
+    Returns
+    -------
+    nii_mtsat: msct_image object
+    nii_t1map: msct_image object
+    """
+    # params
+    nii_t1map = None  # it would be possible in the future to input T1 map from elsewhere (e.g. MP2RAGE). Note: this
+                      # T1map needs to be in s unit.
+    b1correctionfactor = 0.4  # empirically defined in https://www.frontiersin.org/articles/10.3389/fnins.2013.00095/full#h3
+
+    # convert all TRs in s
+    tr_mt *= 0.001
+    tr_pd *= 0.001
+    tr_t1 *= 0.001
+
+    # Convert flip angles into radians
+    fa_mt_rad = math.radians(fa_mt)
+    fa_pd_rad = math.radians(fa_pd)
+    fa_t1_rad = math.radians(fa_t1)
+
+    # ignore warnings from division by zeros (will deal with that later)
+    # note: do not use over='ignore' because it results in wrong T1 maps
+    seterr_old = np.seterr(over='ignore', divide='ignore', invalid='ignore')
```

Comment:
Comment says « do not use over='ignore' because it results in wrong T1 maps » but code does it. Is the comment up-to-date?

Reply:
ah shit ;-) not up to date